hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -76,6 +80,7 @@ public MapQueryEngineImpl(MapServiceContext mapServiceContext) {
         this.partitionService = nodeEngine.getPartitionService();
         this.logger = nodeEngine.getLogger(getClass());
         this.queryResultSizeLimiter = new QueryResultSizeLimiter(mapServiceContext, logger);
+        this.queryOptimizer = QueryOptimizerFactory.newOptimizer(nodeEngine.getGroupProperties());
```

Comment:
Sure. But eventually it is best to inject a QueryOptimizer into the QueryEngine as argument. Anyhow currently not a big deal. Once we move towards DI we can fix this easily